pilot/pkg/networking/grpcgen/grpcgen.go

```diff
@@ -106,34 +109,39 @@ func (g *GrpcConfigGenerator) BuildListeners(node *model.Proxy, push *model.Push
 			}
 			for _, p := range sv.Ports {
 				hp := net.JoinHostPort(shost, strconv.Itoa(p.Port))
-				ll := &xdsapi.Listener{
+				ll := &listener.Listener{
 					Name: hp,
 				}
 
-				ll.Address = &envoycore.Address{
-					Address: &envoycore.Address_SocketAddress{
-						SocketAddress: &envoycore.SocketAddress{
+				ll.Address = &core.Address{
+					Address: &core.Address_SocketAddress{
+						SocketAddress: &core.SocketAddress{
 							Address: sv.Address,
-							PortSpecifier: &envoycore.SocketAddress_PortValue{
+							PortSpecifier: &core.SocketAddress_PortValue{
 								PortValue: uint32(p.Port),
 							},
 						},
 					},
 				}
-				// TODO: for TCP listeners don't generate RDS, but some indication of cluster name.
-				ll.ApiListener = &envoy_config_listener_v2.ApiListener{
-					ApiListener: util.MessageToAny(&v2.HttpConnectionManager{
-						RouteSpecifier: &v2.HttpConnectionManager_Rds{
-							Rds: &v2.Rds{
-								ConfigSource: &envoycore.ConfigSource{
-									ConfigSourceSpecifier: &envoycore.ConfigSource_Ads{
-										Ads: &envoycore.AggregatedConfigSource{},
-									},
+				hcm := &hcm.HttpConnectionManager{
+					RouteSpecifier: &hcm.HttpConnectionManager_Rds{
+						Rds: &hcm.Rds{
+							ConfigSource: &core.ConfigSource{
+								ConfigSourceSpecifier: &core.ConfigSource_Ads{
+									Ads: &core.AggregatedConfigSource{},
 								},
-								RouteConfigName: hp,
 							},
+							RouteConfigName: hp,
 						},
-					}),
+					},
+				}
+				hcmAny := util.MessageToAny(hcm)
+				// TODO: grpc-go still expects the v2 Http connection manager TypeUrl. Fix this when it is changed.
+				// https://github.com/grpc/grpc-go/blob/master/xds/internal/version/version.go#L48.
+				hcmAny.TypeUrl = "type.googleapis.com/envoy.config.filter.network.http_connection_manager.v2.HttpConnectionManager"
+				// TODO: for TCP listeners don't generate RDS, but some indication of cluster name.
```

Comment:
does tcp make sense for gRPC?

Reply:
I just copied the comment from above. But I guess it does not make sense